src/node.rs

```diff
@@ -232,6 +232,18 @@ pub enum Type {
     Audio = 6,
 }
 
+impl FromPrimitive for Type {
+    fn from_i64(n: i64) -> Option<Type> {
+        match n {
+            0...6 => Some(unsafe { ::std::mem::transmute(n as u8) }),
```

Comment:
Not sure if this is correct or not.
I could match for each variant separately, but I didn't feel like typing it all out.